src/register/tests/RegistrationPage.test.jsx

```diff
@@ -774,6 +798,115 @@ describe('RegistrationPage', () => {
         MARKETING_EMAILS_OPT_IN: '',
       });
     });
+
+    // ******** persist state tests ********
+
+    it('should clear form field errors in redux store on onFocus', () => {
+      store.dispatch = jest.fn(store.dispatch);
+      const registrationPage = mount(reduxWrapper(<IntlRegistrationPage {...props} />));
+      registrationPage.find('input#name').simulate('focus');
+      expect(store.dispatch).toHaveBeenCalledWith(registerSetFormData(formData));
+    });
+
+    it('should set username in redux store if usernameSuggestion is clicked', () => {
+      formData = {
+        ...formData,
+        username: 'testname',
+      };
+      store = mockStore({
+        ...initialState,
+        register: {
+          ...initialState.register,
+          usernameSuggestions: ['testname', 't.name', 'test_0'],
+        },
+      });
+      store.dispatch = jest.fn(store.dispatch);
+      const registerPage = mount(reduxWrapper(<IntlRegistrationPage {...props} />));
+      registerPage.find('input#name').simulate('change', { target: { value: 'test name', name: 'name' } });
+      registerPage.find('button.username-suggestion').at(0).simulate('click');
+      expect(store.dispatch).toHaveBeenCalledWith(registerSetFormData(formData));
+    });
+
+    it('should set email in redux store if emailSuggestion is clicked', () => {
+      formData = {
+        ...formData,
+        email: 'test@gmail.com',
+      };
+      store = mockStore({
+        ...initialState,
+        register: {
+          ...initialState.register,
+        },
+      });
+      store.dispatch = jest.fn(store.dispatch);
+      const registrationPage = mount(reduxWrapper(<IntlRegistrationPage {...props} />));
+      registrationPage.find('input#email').simulate('blur', { target: { value: 'test@gmail.con', name: 'email' } });
+      registrationPage.find('RegistrationPage').setState({ emailErrorSuggestion: 'test@gmail.com' });
```

Comment:
Why we are setting `emailErrorSuggestion` state itself? Should it not set on onBlure when user enter incorrect email?